File: backend/app/schemas/wing.py
from pydantic import BaseModel
from typing import List, Optional
from app.schemas.activity import ActivityResponse
from app.schemas.photo import PhotoResponse


class WingBase(BaseModel):
    name: str
    slug: str
    about: str
    vision: str
    mission: str


class WingResponse(WingBase):
    id: int
    
    class Config:
        from_attributes = True


class WingWithRelations(WingResponse):
    activities: List[ActivityResponse] = []
    photos: List[PhotoResponse] = []
    
    class Config:
        from_attributes = True

File: frontend/src/types/wing.ts
import { Activity } from './activity';
import { Photo } from './photo';

export interface Wing {
  id: number;
  name: string;
  slug: string;
  about: string;
  vision: string;
  mission: string;
}

export interface WingWithRelations extends Wing {
  activities: Activity[];
  photos: Photo[];
}
